### PR TITLE
LibJS/Temporal: Move time zone annotation parsing into ParseISODateTime

### PR DESCRIPTION
**Pull request by snooze6214**
This is an editorial change in the temporal spec.

See: tc39/proposal-temporal@c410e25e4761bd9c86880219558d25074d28839f
Ticks of another item in #15525 :^)

### CODE REVIEW

**Review by davidot — APPROVED** [review]
LGTM, just one thing about the commit message, I believe we just use `LibJS: ...` not `LibJS/Subcategory: ...` (At least I can't find any commits with that in git log.

**Comment by snooze6214** [comment]
> LGTM, just one thing about the commit message, I believe we just use `LibJS: ...` not `LibJS/Subcategory: ...` (At least I can't find any commits with that in git log. (But it's fine to me since it does fit)

@davidot alright! I'll remove the temporal from the commit then?

**Comment by davidot** [comment]
> @davidot alright! I'll remove the temporal from the commit then?

Yeah I think so, if anyone wants you to put it back blame me

**Comment by snooze6214** [comment]
> > @davidot alright! I'll remove the temporal from the commit then?
> 
> Yeah I think so, if anyone wants you to put it back blame me

haha, okay! Thanks for reviewing!

**Comment by linusg** [comment]
Lubrsi started doing it, I don't mind either way

File: Userland/Libraries/LibJS/Runtime/Temporal/AbstractOperations.cpp [hunk 1]
```diff
@@ -660,7 +660,7 @@ ThrowCompletionOr<Value> to_relative_temporal_object(VM& vm, Object const& optio
         match_behavior = MatchBehavior::MatchMinutes;
 
         // See NOTE above about why this is done.
-        result = move(parsed_result.date_time);
+        result = move(parsed_result);
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/Temporal/AbstractOperations.cpp:663]: The comment on line 615 is no longer true after this change, and then this one can be removed as well. I think we might just be able to name it `result` in the first place and get rid of this altogether?

**Comment by Lubrsi** [comment]
I think I just put the Temporal subcategory because the commit message was a bit vague by itself

File: Userland/Libraries/LibJS/Runtime/Temporal/AbstractOperations.cpp [hunk 2]
```diff
@@ -1318,23 +1344,15 @@ ThrowCompletionOr<TemporalInstant> parse_temporal_instant_string(VM& vm, String
...
-
-    // 4. Return the Record { [[Year]]: result.[[Year]], [[Month]]: result.[[Month]], [[Day]]: result.[[Day]], [[Hour]]: result.[[Hour]], [[Minute]]: result.[[Minute]], [[Second]]: result.[[Second]], [[Millisecond]]: result.[[Millisecond]], [[Microsecond]]: result.[[Microsecond]], [[Nanosecond]]: result.[[Nanosecond]], [[Calendar]]: result.[[Calendar]], [[TimeZoneZ]]: timeZoneResult.[[Z]], [[TimeZoneOffsetString]]: timeZoneResult.[[OffsetString]], [[TimeZoneName]]: timeZoneResult.[[Name]] }.
-    // NOTE: This returns the two structs together instead of separated to avoid a copy in ToTemporalZonedDateTime, as the spec tries to put the result of InterpretTemporalDateTimeFields and ParseTemporalZonedDateTimeString into the same `result` variable.
-    // InterpretTemporalDateTimeFields returns an ISODateTime, so the moved in `result` here is subsequently moved into ParseTemporalZonedDateTimeString's `result` variable.
-    return TemporalZonedDateTime { .date_time = move(result), .time_zone = move(time_zone_result) };
+    // 2. Return ? ParseISODateTime(isoString).
+    return TRY(parse_iso_date_time(vm, *parse_result));
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/Temporal/AbstractOperations.cpp:1355]: You can omit `TRY` at the end of a function given `parse_iso_date_time` returns the same type - saves us one `ThrowCompletionOr` unwrap.

File: Userland/Libraries/LibJS/Runtime/Temporal/AbstractOperations.cpp [hunk 3]
```diff
@@ -1557,46 +1575,15 @@ ThrowCompletionOr<TemporalMonthDay> parse_temporal_month_day_string(VM& vm, Stri
...
-        z = false;
-
-        // b. Let offsetString be undefined.
-        // c. Let timeZone be undefined.
-    }
-
-    // 5. Return the Record { [[Year]]: result.[[Year]], [[Month]]: result.[[Month]], [[Day]]: result.[[Day]], [[Hour]]: result.[[Hour]], [[Minute]]: result.[[Minute]], [[Second]]: result.[[Second]], [[Millisecond]]: result.[[Millisecond]], [[Microsecond]]: result.[[Microsecond]], [[Nanosecond]]: result.[[Nanosecond]], [[Calendar]]: result.[[Calendar]], [[TimeZoneZ]]: z, [[TimeZoneOffsetString]]: offsetString, [[TimeZoneIANAName]]: timeZone }.
-    return TemporalZonedDateTime { .date_time = move(result), .time_zone = { .z = z, .offset_string = move(offset_string), .name = move(time_zone) } };
+    // 2. Return ? ParseISODateTime(isoString).
+    return TRY(parse_iso_date_time(vm, *parse_result));
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/Temporal/AbstractOperations.cpp:1586]: Likewise

File: Userland/Libraries/LibJS/Runtime/Temporal/ZonedDateTime.cpp [hunk 4]
```diff
@@ -229,16 +229,16 @@ ThrowCompletionOr<ZonedDateTime*> to_temporal_zoned_date_time(VM& vm, Value item
...
+        auto temporal_calendar_like = parsed_result.calendar.has_value()
+            ? js_string(vm, parsed_result.calendar.value())
             : js_undefined();
         calendar = TRY(to_temporal_calendar_with_iso_default(vm, temporal_calendar_like));
 
         // l. Set matchBehaviour to match minutes.
         match_behavior = MatchBehavior::MatchMinutes;
 
         // See NOTE above about why this is done.
-        result = move(parsed_result.date_time);
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/Temporal/ZonedDateTime.cpp:241]: Likewise

**Review by linusg — COMMENTED** [review]
Nicely done! :100:
Couple of small comments.

**Comment by davidot** [comment]
> Lubrsi started doing it, I don't mind either way

Didn't pull before I looked at the log my bad.
I don't mind either way as well. I'll try to remember for any next PRs.

File: Userland/Libraries/LibJS/Runtime/Temporal/AbstractOperations.cpp (hunk 1 above)
Comment (snooze6214) [reply Userland/Libraries/LibJS/Runtime/Temporal/AbstractOperations.cpp:663]: Yep! seems like we can get rid of that `parsed_result` business altogether. I'll make that change asap

File: Userland/Libraries/LibJS/Runtime/Temporal/AbstractOperations.cpp (hunk 2 above)
Comment (snooze6214) [reply Userland/Libraries/LibJS/Runtime/Temporal/AbstractOperations.cpp:1355]: Yeah sure

**Review by linusg — APPROVED** [review]
:yakkiss:

**Pull request merged**